proxy/src/transport/tls/config.rs

```diff
@@ -230,6 +229,30 @@ impl CommonSettings {
     }
 }
 
+impl PathAndHash {
+    fn new(path: PathBuf) -> Self {
+        Self {
+            path,
+            last_hash: RefCell::new(None),
+        }
+    }
+
+    fn has_changed(&self) -> io::Result<bool> {
+        let contents = fs::read(&self.path)?;
+        let hash = Some(digest::digest(&digest::SHA256, &contents[..]));
```

Comment:
thought: we could probably avoid computing the digest if the size of the files has changed.  we only need to compute a digest to ensure they _have not_ changed, not to test whether they _might have_ changed. So we only need to compute the digest when the size and timestamp are the same.

Reply:
That's a good point!